### PR TITLE
Reorganize Docs Navbar and Fill in Gaps

### PR DESCRIPTION
**Pull request by aliabd**
- Reorganizes docs navbar so it make more sense, and removes Combining Interfaces (except for TabbedInterface which is its own page now) 
- Fills in docs gap in Info, Warning, and Error (descriptions, example usages, and demos) and groups them together in the navbar. 
- Makes navbar reactive when user switches between main and pip. 

Closes: #5218 
Closes: #5567 
Closes: #5674

### CODE REVIEW

**Comment by aliabd** [comment]
@osanseviero please take a look and lmk if this is good for Info, Warning, and Error :)

**Comment by aliabd** [comment]
@osanseviero there was an issue with the CI, you can check now

**Comment by abidlabs** [comment]
You'll also need to see visit the `main` version of the docs, e.g. here: https://gradio-e19501bci-hugging-face.vercel.app/main/docs/interface

@aliabd I'll take a look too, would you be able to resolve the conflicts?

**Comment by osanseviero** [comment]
Very nice :fire: this is cool! Maybe for a follow-up for the components: I think with the number of components and with incoming community component, maybe updating https://gradio-e19501bci-hugging-face.vercel.app/main/docs/components to focus on a quick showcase of each component could be worth it, and splitting them by sections.

File: gradio/helpers.py [hunk 1]
```diff
@@ -1114,20 +1114,44 @@ def log_message(message: str, level: Literal["info", "warning"] = "info"):
...
 
 
+set_documentation_group("modals")
+
+
 @document()
 def Warning(message: str = "Warning issued."):  # noqa: N802
     """
-    This function allows you to pass custom warning messages to the user. You can do so simply with `gr.Warning('message here')`, and when that line is executed the custom message will appear in a modal on the demo.
+    This function allows you to pass custom warning messages to the user. You can do so simply with `gr.Warning('message here')`, and when that line is executed the custom message will appear in a modal on the demo. Queue must be enabled to use Warning.
```
Comment (abidlabs) [inline gradio/helpers.py:1123]:
```suggestion
    This function allows you to pass custom warning messages to the user. You can do so simply with `gr.Warning('message here')`, and when that line is executed the custom message will appear in a modal on the demo. The modal is yellow by default and has the heading: "Warning." Queue must be enabled to use Warning.
```

File: gradio/helpers.py [hunk 2]
```diff
@@ -1114,20 +1114,44 @@ def log_message(message: str, level: Literal["info", "warning"] = "info"):
...
+            demo.load(hello_world, inputs=None, outputs=[md])
+        demo.queue().launch()
     """
     log_message(message, level="warning")
 
 
 @document()
 def Info(message: str = "Info issued."):  # noqa: N802
     """
+    This function allows you to pass custom info messages to the user. You can do so simply with `gr.Info('message here')`, and when that line is executed the custom message will appear in a modal on the demo.  Queue must be enabled to use Info.
```
Comment (abidlabs) [inline gradio/helpers.py:1143]:
```suggestion
    This function allows you to pass custom info messages to the user. You can do so simply by writing `gr.Info('message here')` in your function, and when that line is executed the custom message will appear in a (gray, by default) modal on the demo.  Queue must be enabled to use Info.
```

**Review by abidlabs — APPROVED** [review]
Looks great @aliabd! I tweaked the docstrings just a bit to differentiate between `gr.Info` and `gr.Warning`

**Pull request merged**